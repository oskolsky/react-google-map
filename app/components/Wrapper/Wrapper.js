import React from 'react'
import PropTypes from 'prop-types'

import { Map } from 'components'

import styles from './Wrapper.scss'

export default class Wrapper extends React.Component {
    static propTypes = {
        location: PropTypes.shape({
            coords: PropTypes.shape({
                latitude: PropTypes.number.isRequired,
                longitude: PropTypes.number.isRequired,
            }).isRequired,
        }).isRequired,
        getLocation: PropTypes.func.isRequired,
    }

    componentWillMount() {
        this.props.getLocation()
    }

    render() {
        return (
            <div className={styles.base}>
                <div className={styles.map}>
                    <Map center={this.props.location.coords} />
                </div>
            </div>
        )
    }
}
