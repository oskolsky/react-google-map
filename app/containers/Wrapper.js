import { connect } from 'react-redux'

import { Wrapper } from 'components'
import { getLocation } from 'actions/Location'
import { openDialog } from 'actions/Dialog'

const mapStateToProps = (state) => {
    return {
        location: state.location,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getLocation: () => dispatch(getLocation()),
        onOpenDialog: (content) => dispatch(openDialog(content)),
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Wrapper)
