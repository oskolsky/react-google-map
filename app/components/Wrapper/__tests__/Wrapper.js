import React from 'react'
import { shallow } from 'enzyme'

import Wrapper from '../Wrapper'

function setup(options) {
    const props = {
        location: {
            coords: {
                latitude: 0,
                longitude: 0,
            },
        },
        getLocation: jest.fn(),
        onOpenDialog: jest.fn(),
    }

    const enzymeWrapper = shallow(
        <Wrapper {...props} {...options} />
    )

    return {
        props,
        enzymeWrapper,
    }
}

let enzymeWrapper = null

describe('Wrapper specification', () => {
    afterEach(() => {
        enzymeWrapper.unmount()
    })

    it('renders correctly', () => {
        enzymeWrapper = setup().enzymeWrapper

        expect(enzymeWrapper).toMatchSnapshot()
    })
})
