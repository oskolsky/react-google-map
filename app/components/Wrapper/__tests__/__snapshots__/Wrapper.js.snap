// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper specification renders correctly 1`] = `
<div
  className="base"
>
  <div
    className="map"
  >
    <Map
      center={
        Object {
          "latitude": 0,
          "longitude": 0,
        }
      }
      zoom={10}
    />
  </div>
</div>
`;
